lib/saml_idp/request.rb

```diff
@@ -71,6 +71,38 @@ def requested_authn_context
       end
     end
 
+    def requested_authn_contexts
+      if authn_request? && authn_context_nodes.length > 0
+        authn_context_nodes.map(&:content)
+      else
+        nil
+      end
```

Comment:
what if this always returned an array, including an empty array? Then the methods that reference it could do a no-op `.each`